collector/tcpstat.go

```diff
@@ -0,0 +1,210 @@
+// +build !notcpstat
+
+package collector
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	procTcpStat  = "/proc/net/tcp"
```

Comment:
@matthiasr Yeah, the only problem is that it might accidentally really slow down people's node exporters or machines especially in high-load situations (tens of thousands of connections).